rhobs/recording/build_service_availability_recording_rules.yaml

```diff
@@ -0,0 +1,15 @@
+# Metric format needed
+# redhat_appstudio_buildservice_global_github_app_available(service="build-service-controller-manager-metrics-service") -> konflux_up(service="build-service-controller-manager-metrics-service", check="github")
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-build-service-availability
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+    - name: build_service_github_availability
+      interval: 1m
+      rules:
+        - record: konflux_up
+          expr: label_replace(redhat_appstudio_buildservice_global_github_app_available, "check", "github", "","")
```

Comment:
I don't see the baseline metrics in appSRE Grafana.
The recording rule will not generate any time series until those metrics are there.

Reply:
You mentioned `redhat_appstudio_buildservice_global_github_app_available` and the other one to be existing. I don't see them on AppSRE Grafana, so the recording rule will not be able to generate `konflux_up` time series for them in AppSRE Grafana.

Where do we have those metrics at the moment?

Reply:
So far they're just exported from the services.

Reply:
alright. Do you want this PR merged prior to having those metrics in AppSRE Grafana? (if that is the case then the alert rules I mentioned should only be added after the metrics are present in AppSRE Grafana)

Reply:
If services export them, then a service monitor should be created on the clusters. This will make Prometheus on the cluster scrape those metrics.

This should be enough for the metrics to be collected by Prometheus and remote-written to RHOBS, and then they'll be displayed in AppSRE Grafana.

Reply:
@amisstea will be able to provide a reference to a service monitor?

Reply:
@mshaposhnik, in case you're looking for it, documentation about service monitors is in [infra-deployments](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/monitoring#2-service-monitors). That being said, I do see the `redhat_appstudio_buildservice_global_github_app_available` metric in AppSRE [grafana](https://grafana.app-sre.devshift.net/explore?schemaVersion=1&panes=%7B%22xnx%22:%7B%22datasource%22:%22P22466E8E7855F1E0%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22redhat_appstudio_buildservice_global_github_app_available%22,%22range%22:true,%22instant%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PDCCF087A30F0737B%22%7D,%22editorMode%22:%22code%22,%22legendFormat%22:%22__auto%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) for our stage and prod tenants so it would appear the existing build ServiceMonitor is working.

Reply:
![image](https://github.com/redhat-appstudio/o11y/assets/1651062/0f72b1f5-27fe-4b06-ada0-4f098f303511)

Reply:
build service can be seen already, image controller not yet - probably it is just not yet rolled to stg

Reply:
I'm off today and tomorrow, so I cannot merge it myself at the moment, but I have no objections for having it merged.